src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(last, value,null);
+        if (size == 0) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        validIndex(index);
+        if (index == 0) {
+            Node<T> currentNode = new Node<>(null, value, first);
+            first.prev = currentNode;
+            first = currentNode;
+        } else {
+            Node<T> oldNode = findNode(index);
+            Node<T> currentNode = new Node<>(oldNode.prev, value, oldNode);
+            oldNode.prev.next = currentNode;
+            oldNode.prev = currentNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validIndex(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        validIndex(index);
+        Node<T> oldNode = findNode(index);
+        T oldNodeValue = oldNode.item;
+        oldNode.item = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validIndex(index);
+        Node<T> oldNode = findNode(index);
+        T oldNodeItem = oldNode.item;
+        unlink(oldNode);
```

Comment:
Recommendation. You can return an item in the unlink() method. It will save several lines.
It may look like this, return unlink(findNode(index));
So, you can change it if you want,  the last word for the mentor.